api/proto/teleport/okta/v1/okta_service.proto

```diff
@@ -51,8 +51,166 @@ service OktaService {
+  }
+  // List of groups that match the filter criteria.
+  repeated Group groups = 1;
+}
+
+// EnrollIntegrationRequest is the request message for enrolling a new Okta integration.
+message EnrollIntegrationRequest {
+  // URL of the Okta organization.
+  string okta_organization_url = 1;
+  // Credentials to access the Okta API.
+  OktaAPICredentials api_credentials = 2;
+  // SCIM token for integration.
+  string scim_token = 3;
+  // Enable synchronization of access lists.
+  bool enable_access_list_sync = 4;
+  // Enable user synchronization.
+  bool enable_user_sync = 5;
+  // Enable synchronization of app groups.
+  bool enable_app_group_sync = 6;
+  // Settings related to access list synchronization.
+  AccessListSettings access_list_settings = 7;
+  // Option to reuse an existing connector.
+  string reuse_connector = 8;
+  // SSO metadata URL for integration.
+  string sso_metadata_url = 9;
+}
+
+// UpdateIntegrationRequest is the request message for updating an existing Okta integration.
+message UpdateIntegrationRequest {
+  // Credentials to access the Okta API.
+  OktaAPICredentials api_credentials = 2;
+  // SCIM token for integration.
+  string scim_token = 3;
+  // Enable synchronization of access lists.
+  bool enable_access_list_sync = 4;
+  // Enable user synchronization.
+  bool enable_user_sync = 5;
+  // Enable synchronization of app groups.
+  bool enable_app_group_sync = 6;
+  // Settings related to access list synchronization.
+  AccessListSettings access_list_settings = 7;
+  // Option to reuse an existing connector.
+  string reuse_connector = 8;
+  // SSO metadata URL for integration.
+  string sso_metadata_url = 9;
 }
 
+// AccessListSettings contains the settings for access list synchronization.
+message AccessListSettings {
+  // Filters for groups to include in the access list.
+  repeated string group_filters = 2;
+  // Filters for apps to include in the access list.
+  repeated string app_filters = 3;
+  // Default owner for access list items.
+  repeated string default_owner = 4;
+}
+
+// OktaAPICredentials provides the authentication details for accessing Okta API.
+message OktaAPICredentials {
+  oneof auth {
+    // OAuth ClientID for authentication.
+    string oauth_id = 1;
+    // SSM Bearer token for authentication.
+    string ssm_bearer_token = 2;
+  }
+}
+
+// EnrollIntegrationResponse is the response message for enrolling a new Okta integration.
+message EnrollIntegrationResponse {
+  // Plugin information.
+  types.PluginV1 plugin = 1;
+  // Details of the connector.
+  ConnectorInfo connector_info = 2;
+}
+
+// UpdateIntegrationResponse is the request for listing paginated Okta import rules.
+message UpdateIntegrationResponse {
+  // Plugin information.
+  types.PluginV1 plugin = 1;
+  // Details of the connector.
+  ConnectorInfo connector_info = 2;
+}
+
+// ConnectorInfo contains information about the connector.
+message ConnectorInfo {
+  // Okta application ID.
+  string okta_app_id = 1;
```

Comment:
nit: `okta_` reads redundant but I lack context on how useful that prefix will be during implementation.